src/controllers/StudentController.js

```diff
@@ -0,0 +1,176 @@
+import { z } from 'zod';
+import { fromZodError } from 'zod-validation-error';
+
+import { EntryExists } from '../errors/EntryExists.js';
+
+import StudentService from '../services/StudentService.js';
+
+async function list(request, response) {
+  if (request.role == 'STUDENT') {
```

Comment:
Por segurança, caso no futuro venha a ser criada outra role (ainda que improvável), é preferível checar se a role do usuário está dentro das roles permitidas.

```suggestion
  if (!['ADMIN', 'VERIFIER'].includes(request.role)) {
```